Profile Page: add servlet

ElleTojaroon opened the pull request:
Added an empty profile page to navigate by /profile

ElleTojaroon commented:
Second commit: I added profile.jsp

<img width="1231" alt="screen shot 2018-03-24 at 12 15 37 am" src="https://user-images.githubusercontent.com/15943485/37860284-83bc46d6-2ef8-11e8-8765-3e540d34ce38.png">

ElleTojaroon commented:
3rd commit: navigation tab on every page has "My Profile" that links to the logged in user's profile.

<img width="1233" alt="screen shot 2018-03-24 at 4 22 10 am" src="https://user-images.githubusercontent.com/15943485/37861938-2a043c7a-2f1b-11e8-9310-7280556302eb.png">

ElleTojaroon commented:
Just uploaded my final commit for the first draft. Here're the screenshot of how it looks:

<img width="1234" alt="screen shot 2018-03-30 at 2 09 12 am" src="https://user-images.githubusercontent.com/15943485/38126576-83b5e1a8-33bf-11e8-8dd2-4ea9a7ee1ced.png">
<img width="1237" alt="screen shot 2018-03-30 at 2 09 23 am" src="https://user-images.githubusercontent.com/15943485/38126611-ab8b4920-33bf-11e8-8cc8-711b4ec71331.png">

mynguyenbmc commented on src/test/java/codeu/model/store/basic/UserStoreTest.java:
```diff
@@ -84,6 +85,55 @@ public void testIsUserRegistered_false() {
+  public void testUpdateUserDescription_updateSucessful() {
+    String newDescription = "new description";
+    Mockito.when(
+        mockPersistentStorageAgent.updateUserDescription(USER_ONE.getName(), newDescription))
```
just for my own understanding: What is USER_ONE?

ElleTojaroon replied:
fake user object defined in UserStoreTest.java:20
In intellij, you can cmd+click or cmd+b on a symbol to go to its declaration.

mynguyenbmc commented on src/main/java/codeu/model/data/User.java:
```diff
@@ -15,6 +15,9 @@
 import java.time.Instant;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
```
I don't see these used here?

ElleTojaroon replied:
Done.

mynguyenbmc commented:
I don't know why Github says there's a conflict in PersistentDataStore when it seems totally reasonable to me. I also have several minor questions as noted above. Otherwise, everything LGTM.
For important changes like this, however, it might be better to have approvals from more than 1 person. Thank you so much Elle for doing all of this though <3!

ElleTojaroon commented:
@mynguyenbmc There's a conflict because I added a few lines and also modified a line in the code. I'll resolve the conflict once I get more LG. Thanks for reviewing!

krtejeda commented:
I completely forgot to press send for these pending comments. If you get the chance I'd appreciate a response though. Sorry!

ElleTojaroon commented on src/main/java/codeu/controller/ProfileServlet.java:
```diff
@@ -0,0 +1,157 @@
+      }
+
+      User user = userStore.getUser(logInName);
+      if (user == null) {
```
Just in case getUser() returns null, so we won't get RuntimeException

ElleTojaroon commented on src/main/java/codeu/controller/ProfileServlet.java:
```diff
@@ -0,0 +1,157 @@
+        String ownerName = requestUrl.substring("/profile/".length());
+        User owner = userStore.getUser(ownerName);
+        if (owner == null) {
+            // owner was not found, direct to homepage
```
Good point! lemme check.

ElleTojaroon commented on src/main/java/codeu/model/store/persistence/PersistentStorageAgent.java:
```diff
@@ -17,7 +17,6 @@
 import codeu.model.data.Conversation;
 import codeu.model.data.Message;
 import codeu.model.data.User;
-import codeu.model.store.persistence.PersistentDataStore;
```
It's not needed. I just have auto import optimized setup.
